WordPress/Classes/ViewRelated/NUX/SigninSelfHostedViewController.swift

```diff
@@ -4,9 +4,8 @@ import WordPressShared
 /// Provides a form and functionality to sign-in and add an existing self-hosted 
 /// site to the app.
 ///
-@objc class SigninSelfHostedViewController : NUXAbstractViewController, SigninKeyboardResponder
+@objc class SigninSelfHostedViewController : NUXAbstractViewController, SigninKeyboardResponder, SigninWPComSyncHandler
```

Comment:
> Maybe its just naming confusion? 

I think naming is part of it, but also the architecture. By conforming to `SigninWPComSyncHandler`, the self hosted VC now has a `syncWPCom` method.

Rethinking this a bit, what I'd try: Instead of a protocol extension, move `syncWPCom` to a separate object and use a delegate. It feels like this is what `AccountServiceFacade` should be. 

This sounds like taking it too far for the scope of this PR, but I wanted to leave a note so we can revisit this later.

> we're only offering the option to sign in with saved credentials to wpcom sites so we're not presenting the option for self-hosted users

I meant if the URL entered was for a wordpress.com site, and we ended up doing a wp.com sign in. But yeah, separate issue.